### PR TITLE
chore: update proto version and enable grpc gzip

### PR DESCRIPTION
**Pull request by paomian**

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -36,22 +37,74 @@ pub struct Client {
     inner: Arc<Inner>,
 }
 
+#[derive(Default)]
+pub struct ClientBuilder {
```
Comment (shuiyisong) [inline src/client.rs:41]: Can we use [derive_builder](https://crates.io/crates/derive_builder) to reduce boilerplate code?

Comment (paomian) [reply src/client.rs:41]: [derive_builder](https://crates.io/crates/derive_builder) will return Result. but we don't need it.

File: src/client.rs [hunk 2]
```diff
@@ -36,22 +37,74 @@ pub struct Client {
...
+            inner: Arc::new(inner),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum Compression {
+    Gzip,
+    Zstd,
+    Plain,
```
Comment (sunng87) [inline src/client.rs:90]: Rename to `None`

File: src/client.rs [hunk 3]
```diff
@@ -63,51 +116,93 @@ impl Inner {
...
+            .channel_manager(channel_manager)
+            .peers(normailze_urls(urls))
+            .build();
+
         Self {
             inner: Arc::new(inner),
         }
     }
 
+    #[deprecated(since = "0.1.0", note = "should be removed in the future")]
```
Comment (sunng87) [inline src/client.rs:199]: Add description for what we recommend to use just like `with_manager_and_urls`

**Comment by sunng87** [comment]
looks almost good to me

**Review by sunng87 — APPROVED** [review]

File: Cargo.toml [hunk 4]
```diff
@@ -10,15 +10,16 @@ dashmap = "5.4"
...
 parking_lot = "0.12"
 prost = "0.12"
 rand = "0.8"
 snafu = "0.7"
 tokio = { version = "1", features = ["rt", "time"] }
 tokio-stream = { version = "0.1", features = ["net"] }
-tonic = { version = "0.10", features = ["tls", "tls-roots"] }
+tonic = { version = "0.11", features = ["tls", "tls-roots", "gzip", "zstd"] }
 tower = "0.4"
+derive_builder = "0.20.0"
```
Comment (shuiyisong) [inline Cargo.toml:22]:
```suggestion
derive_builder = "0.20"
```

Remove the patch version.

File: src/client.rs [hunk 5]
```diff
@@ -140,6 +219,17 @@ impl Client {
     }
 }
 
+fn normailze_urls<U, A>(urls: A) -> Vec<String>
```
Comment (shuiyisong) [inline src/client.rs:222]: typo: should be `normalize_urls`.

**Review by shuiyisong — APPROVED** [review]

**Review by shuiyisong — APPROVED** [review]
🚀

File: src/client.rs [hunk 6]
```diff
@@ -36,22 +38,83 @@ pub struct Client {
...
+    Gzip,
+    Zstd,
+    None,
+}
+
+impl Default for Compression {
+    fn default() -> Self {
+        Self::Gzip
+    }
+}
```
Comment (fengys1996) [inline src/client.rs:99]: Nit: 
```suggestion
#[derive(Debug, Clone, Default)]
pub enum Compression {
    #[default]
    Gzip,
    Zstd,
    None,
}
```

**Review by fengys1996 — APPROVED** [review]
LGTM

**Pull request merged**